theano/gof/cmodule.py

```diff
@@ -2310,5 +2319,314 @@ def print_command_line_error():
             return dlimport(lib_filename)
 
 
-def icc_module_compile_str(*args):
-    raise NotImplementedError()
+class ICC_compiler(Compiler):
+    # The equivalent flags of --march=native used by icpc.
+    march_flags = None
+
+    # TODO: Can't check AMD CPU
+    supports_amdlibm = True
+
+    @staticmethod
+    def version_str():
+        icc_ver = [int(n) for n in icc_version_str.split('.')[:2]]
+        if bool(icc_ver < [16, 0]):
+            _logger.warn(
+                "WARNING: Your Intel Compiler Version is %s. "
+                "For better performance, we suggest you use the latest version. " % icc_version_str)
+            _logger.info(
+                "WARNING: Your Intel Compiler Version is %s. "
+                "For better performance, we suggest you use the latest version. " % icc_version_str)
```

Comment:
don't log twice. I think the warn level is good. It should appear by default like that.